### PR TITLE
feat: improve ens caching of treasury votes table

### PR DESCRIPTION
**Pull request by ECWireless**
Co-authored-by: Rick Staa <rick.staa@outlook.com>

## Description

Module-level cache to avoid repeated ENS lookups across renders/navigations.

## Type of Change

- [x] feat: New feature
- [ ] fix: Bug fix
- [ ] docs: Documentation update
- [ ] style: Code style/formatting changes (no logic changes)
- [ ] refactor: Code refactoring (no behavior change)
- [ ] perf: Performance improvement
- [ ] test: Adding or updating tests
- [ ] build: Build system or dependency changes
- [ ] ci: CI/CD changes
- [ ] chore: Other changes

## Related Issue(s)

Related: #539

^ closes and replaces the above PR

## Changes Made

- Adds memory caching for ens lookups in the `TreasuryVoteTable`
- Includes `ENS_CACHE_MAX_ENTRIES` for edge case memory bloating
- Includes in-flight dedupes in case the same 2 address get looked up at the same time

## Testing

- [x] Tested locally
- [ ] Added/updated tests
- [x] All tests passing

### How to test (optional unless test is not trivial)

- Check the voting history of a proposal on a treasury proposal page; ensure ens names load quickly

## Impact / Risk

Risk level: Low

Impacted areas: UI

User impact: faster ens name loads

Rollback plan: hotfix or Vercel rollback

## Screenshots / Recordings (if applicable)

NA

## Additional Notes

NA

### CODE REVIEW

File: components/Treasury/TreasuryVoteTable/index.tsx [hunk 1]
```diff
@@ -10,6 +10,58 @@ import TreasuryVotePopover from "./TreasuryVotePopover";
...
+// Module-level cache to avoid repeated ENS lookups across renders/navigations.
+// Bound size keeps memory usage predictable during long sessions.
+const ENS_CACHE_MAX_ENTRIES = 2000;
+const ensCache = new Map<string, string>();
+const ensLookupInFlight = new Map<string, Promise<string>>();
+
+const getCachedEns = (address: string) => {
+  const cached = ensCache.get(address);
+  if (!cached) return undefined;
+
```
[the remark this answers is not in the document]
Comment (ECWireless) [reply components/Treasury/TreasuryVoteTable/index.tsx]: Similar to above comment

**Review by rickstaa — APPROVED** [review]
@ECWireless looks good to me thanks. I however had two suggestions https://github.com/livepeer/explorer/pull/569/changes.
